app/graphql/types/rock_and_pebble_type.rb

```diff
@@ -9,12 +9,20 @@ class RockAndPebbleType < Types::BaseObject
 
     def my_rocks
       user = User.find(object.pebble_id)
-      user.rocks
+      User.active_rocks(user.id)
     end
 
     def my_pebbles
       user = User.find(object.rock_id)
-      user.pebbles
+      User.active_pebbles(user.id)
```

Comment:
(See comment below on `user.rb`)

Reply:
If you decide to keep `active_pebbles` as a User class method, I think you could shorten this to skip the User lookup on line 16 for a small optimization:

```
def my_rocks
  User.active_rocks(object.pebble_id)
end

def my_pebbles
  User.active_pebbles(object.rock_id)
end
```